.github/workflows/build.yml

```diff
@@ -37,19 +43,100 @@ jobs:
           path: '${{ github.workspace }}/backstop/results/'
 
       - name: Host test results on gh pages
-        if: always()
+        if: github.ref != 'refs/heads/main'
         uses: JamesIves/github-pages-deploy-action@4.1.3
         with:
           branch: gh-pages
           folder: ${{ github.workspace }}/backstop/results
           target-folder: ${{ github.event.number }}
+          single-commit: true
+          git-config-name: github-actions[bot]
+          git-config-email: github-actions[bot]@users.noreply.github.com
 
       - name: Post gh pages link under Checks
-        uses: LouisBrunner/checks-action@v1.1.2
-        if: always()
-        with:
-          token: ${{ secrets.GITHUB_TOKEN }}
-          name: Test results
-          conclusion: ${{ job.status }}
-          output: |
-            {"summary": "See backstop html report at https://itwin.github.io/iTwinUI/${{ github.event.number }}/html_report/"}
+        if: github.ref != 'refs/heads/main'
+        uses: actions/github-script@v4
+        with:
+          script: |
+            github.checks.create({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              head_sha: '${{ github.event.pull_request.head.sha }}',
+              conclusion: '${{ job.status }}',
+              name: 'Test results',
+              output: {
+                "title": "Test results",
+                "summary": "See backstop html report at https://itwin.github.io/iTwinUI/${{ github.event.number }}/html_report/"
+              }
+            })
+
+  deploy:
+    name: Deploy demo html pages
```

Comment:
I was thinking of making a separate workflow file for this job but it might be better this way because it's similar to the build job (they use the same actions and everything). Also it could get confusing if there is a "deploy" and another "publish" (for npm release).

---

Other than that, this PR should be good. This is what the gh-pages branch looks like after a fresh deployment: https://github.com/iTwin/iTwinUI/tree/3c5a080c1e10aabb73e1e505bb0dd58a807a94fb